app.rb

```diff
@@ -0,0 +1,145 @@
+require_relative 'person'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+def list_all_books(books)
+  puts 'List of books:'
+  puts
+  books.each_with_index do |book, index|
+    puts "#{index + 1}. Title: #{book.title}, Author: #{book.author}"
+    puts
+  end
+end
+
+def list_all_people(people)
+  puts 'List of people:'
+  puts
+  people.each_with_index do |person, index|
+    person_type = person.instance_of?(Student) ? '[Student]' : '[Teacher]'
+    puts "#{index + 1}. #{person_type} Name: #{person.name}, ID: #{person.id}"
+    puts
+  end
+end
+
+def create_person(people)
+  puts 'Select person type:'
+  puts '1. Student'
+  puts '2. Teacher'
+  print 'Choice: '
+  type = gets.chomp.to_i
+  case type
+  when 1
+    create_student(people)
+  when 2
+    create_teacher(people)
+  else
+    puts 'Invalid option!'
+  end
+end
+
+def create_student(people)
+  print 'Age: '
+  age = gets.chomp.to_i
+  print 'Name: '
+  name = gets.chomp
+  print 'Has parent permission? (Y/N): '
+  permission = gets.chomp == 'Y'
+  student_classroom = Classroom.new('Classroom 1')
+  person = Student.new(student_classroom, age, name: name, parent_permission: permission)
+  people.push(person)
+  puts 'Person created successfully!'
+  puts
+end
+
+def create_teacher(people)
+  print 'Age: '
+  age = gets.chomp.to_i
+  print 'Name: '
+  name = gets.chomp
+  print 'Specialization: '
+  specialization = gets.chomp
+  person = Teacher.new(age, name, specialization)
```

Comment:
- [ ] You did a great job implementing the changes. :+1: When trying to create a teacher the system is raising `Wrong number of arguments` Error. I think this is happening because we have optional parameters. Please try the following to fix the issue. 🧐️

![image](https://github.com/Lornakaboro/School-library/assets/17243325/2b7bb621-b5c1-4353-ab6b-d9b560e0e9d7)